identity/webapp/src/utility/api-caller.ts

```diff
@@ -30,7 +33,7 @@ export async function callRemoteApi(
     validateStatus: (status: number) => status >= 200 && status < 300,
   };
 
-  if (authenticate) {
+  if (authenticate && isAuthenticated(contextState)) {
```

Comment:
Would something that explains the types here be more useful?

```TS
type MaybeUserContextState = {user?: { accessToken?: string }}
type AuthenticatedUserContextState = {user: { accessToken: string }}

const isAuthenticatedContextState = (contextState: ContextState): contextState is AuthenticatedContextState =>
  !!contextState?.user?.accessToken;
```

The more I read this, the odder it is - I wonder if we can pull this into two methods of `callRemoteApi` and `callAuthenticatedRemoteApi` to make it simpler to read?

Reply:
Will see if I can do shinier types here, good idea.

I disagree with the latter part - the job of this module is to transparently proxy API calls, including credentials if they're present, so I would say that it's the job of the remote API - not the caller - to determine whether credentials are necessary/valid/sufficiently scoped for a given call.

Reply:
It seems a little leaky and not massively tranparent as most places I can see calling this code knows the type of the application state (either `UserState` or `ApplicationState`) and whether to authenticate ot not (hence the `authenticate` property).

The `if (authenticate...` part of the method really does make it lok like two methods.

Reply:
I think I'm going to leave this for now because the various `State` and `Context` things aren't quite right here and I think the API routing needs a more holistic refactor - I see what you mean now about the separate methods, but I think the solution is to correctly abstract away the separated logic, rather than making the (unnecessary) distinction more explicit.